apps/src/code-studio/progressRedux.js

```diff
@@ -212,10 +212,16 @@ export const setIsSummaryView = isSummaryView => ({ type: SET_IS_SUMMARY_VIEW, i
 // Do we have one or more lockable stages
 export const hasLockableStages = state => state.stages.some(stage => stage.lockable);
 
-export const lessonNames = state => state.stages.map(stage => stage.name);
 export const hasGroups = state => Object.keys(categorizedLessons(state)).length > 1;
 
-// TODO - account for locked levels
+// Extract the relevant portions of a particular lesson/stage from the store.
+// Note, that this does not include levels
+const lessonFromStage = stage => ({
+  name: stage.name,
+  id: stage.id
+});
+export const lessons = state => state.stages.map(lessonFromStage);
```

Comment:
Hard to say if this is better, but could be expressed as a [`_.pick`](https://lodash.com/docs/4.17.4#pick) operation:

```javascript
const lessonFromStage = stage => _.pick(stage, ['name', 'id']);
```